theme/icons.go

```diff
@@ -633,7 +649,7 @@ func NewSuccessThemedResource(src fyne.Resource) *ThemedResource {
 }
 
 // NewThemedResource creates a resource that adapts to the current theme setting.
```

Comment:
There is an issue with ThemedResource and SVG caching that this will introduce. SVGs are cached by resource name, and themed resource prepends the theme color *name* to the base resource name. This worked when there was only one color for any given color name, and the entire cache was cleaned out on theme variant changes. But now we can have multiple colors for a given theme color name, because we can have multiple active themes throughout the app.

Perhaps we need to introduce a lower-level item like ColoredResource (it doesn't need to be part of the public API, at least initially), that is like ThemedResource but it is parametrized by an actual color.Color value rather than a theme color name, and ThemedResource is a higher-level wrapper of ColoredResource.

Reply:
Good catch.
I think the solution is one that would solve both of your points. To have the resource cached to the scope of an override container.
I cannot see how right now, but that should fix the scope change with the minimal change outside this feature... I hope!